core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/template/TemplateValidator.java

```diff
@@ -157,12 +159,22 @@ private void validateVolumeType(VolumeTemplate value, Platform platform, Validat
         }
     }
 
+    private boolean isIDBrokerInstanceGroup(InstanceGroup instanceGroup) {
+        return GROUP_NAME_ID_BROKER.equalsIgnoreCase(instanceGroup.getGroupName());
+    }
+
     private void validateVolumeCount(VolumeTemplate value, VmType vmType, VolumeParameterType volumeParameterType,
-        ValidationResult.ValidationResultBuilder validationBuilder) {
+        ValidationResult.ValidationResultBuilder validationBuilder, InstanceGroup instanceGroup) {
         if (vmType != null && needToCheckVolume(volumeParameterType, value.getVolumeCount())) {
             VolumeParameterConfig config = vmType.getVolumeParameterbyVolumeParameterType(volumeParameterType);
             if (config != null) {
-                if (value.getVolumeCount() > config.maximumNumber()) {
+                if (isIDBrokerInstanceGroup(instanceGroup)) {
+                    if (value.getVolumeCount() > config.maximumNumber()) {
+                        // IDBroker does not use data volume, so its volume count should be zero
+                        // To be backward-compatible, we only check max limit and allow the min limit to be zero for IDBroker
+                        validationBuilder.error(String.format("Max allowed volume count for '%s': %s", vmType.value(), config.maximumNumber()));
+                    }
+                } else if (value.getVolumeCount() > config.maximumNumber()) {
                     validationBuilder.error(String.format("Max allowed volume count for '%s': %s", vmType.value(), config.maximumNumber()));
                 } else if (value.getVolumeCount() < config.minimumNumber()) {
```

Comment:
Insert `!isIDBrokerInstanceGroup(instanceGroup) && ` at the beginning of condition.